ydb/core/persqueue/partition.cpp

```diff
@@ -1640,24 +1674,26 @@ bool TPartition::ProcessUserActionOrTransaction(TTransaction& t,
 
     if (t.Tx) {
         t.Predicate = BeginTransaction(*t.Tx, ctx);
-
-        ctx.Send(Tablet,
-                 MakeHolder<TEvPQ::TEvTxCalcPredicateResult>(t.Tx->Step,
-                                                             t.Tx->TxId,
-                                                             Partition,
-                                                             *t.Predicate).Release());
-
+        if (t.Tx->SupportivePartitionActor && t.Predicate.GetOrElse(true)) {
```

Comment:
BeginTransaction will set t.Predicate to true in case of Tx->SupportivePartitionActor. Always. Why do we check t.Predicate here?